operations/Makefile

```diff
@@ -50,9 +80,32 @@ tf-03-persistent: ## Preconfigures Terraform persistent environment module
 tf-04-app: ## Preconfigures Terraform app environment module
 	${MAKE} tf-cli TF_STAGE="04-app"
 
+tf-validate: ## Validates the Terraform configuration across all modules
+	${MAKE} tf-cmd TF_STAGE="01-network" TF_CMD="tf validate"
+	${MAKE} tf-cmd TF_STAGE="02-config" TF_CMD="tf validate"
+	${MAKE} tf-cmd TF_STAGE="03-persistent" TF_CMD="tf validate"
+	${MAKE} tf-cmd TF_STAGE="04-app" TF_CMD="tf validate"
+
+tf-plan: ## Outputs the text plan for all modules in /app/plan
+	${MAKE} tf-cmd TF_STAGE="01-network" TF_CMD="tf plan-file"
+	${MAKE} tf-cmd TF_STAGE="02-config" TF_CMD="tf plan-file"
+	${MAKE} tf-cmd TF_STAGE="03-persistent" TF_CMD="tf plan-file"
+	${MAKE} tf-cmd TF_STAGE="04-app" TF_CMD="tf plan-file"
+
 
 # AZURE
 
+# Internal target: set azure service principal values
+.az-configure-service-principal:
+ifdef AZ_SERVICE_PRINCIPAL
+ARM_CLIENT_ID := $(shell echo '$(AZ_SERVICE_PRINCIPAL)' | python3 -c "import sys, json; print(json.load(sys.stdin)['clientId'])")
+ARM_CLIENT_SECRET := $(shell echo '$(AZ_SERVICE_PRINCIPAL)' | python3 -c "import sys, json; print(json.load(sys.stdin)['clientSecret'])")
+ARM_SUBSCRIPTION_ID := $(shell echo '$(AZ_SERVICE_PRINCIPAL)' | python3 -c "import sys, json; print(json.load(sys.stdin)['subscriptionId'])")
+ARM_TENANT_ID := $(shell echo '$(AZ_SERVICE_PRINCIPAL)' | python3 -c "import sys, json; print(json.load(sys.stdin)['tenantId'])")
```

Comment:
Python was chosen to parse the JSON since it's already installed on most machines. Using the dependency `jq` would be cleaner, but I didn't want to force a new dependency.